src/filesystem/drive/inner.rs

```diff
@@ -23,13 +27,19 @@ pub(crate) struct InnerDrive {
 
     nodes: Slab<Node>,
     permanent_id_map: HashMap<PermanentId, NodeId>,
+
+    dirty_nodes: Vec<NodeId>,
```

Comment:
It's a pity we can't rely on a dirty status on an individual Node here... I think NodeId is probably fine for a reference here as this dirty status doesn't persist through serializations but we'll want to make sure the IDs are removed when a node is.

You probably also want to use a HashSet here to ensure the contents are deduplicated.

Reply:
For large filesystems this should be much more efficient then asking every single node if they are dirty, that was the main reason I didn't walk down the path of using the dirty status that nodes carry internally.

For the deduplication thing and removing IDs as nodes are removed I'll dig in more below where the actual cleaning happens